tests/osquery/test_api_views.py

```diff
@@ -143,6 +162,253 @@ def put_json_data(self, url, data, include_token=True):
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), {
+            "platforms": ["darwin", "windows"],
+            "updated_at": atc.updated_at.isoformat(),
+            "columns": ["un", "deux"],
+            "id": atc.id,
+            "created_at": atc.created_at.isoformat(),
+            "table_name": atc.table_name,
+            "query": "select 1 from yo;",
+            "description": atc.description,
+            "path": "/home/yolo",
+            "name": atc.name
+        })
+
+    # update atc
+
+    def test_update_atc_unauthorized(self):
+        response = self.put_json_data(reverse("osquery_api:atc", args=[1]), {}, include_token=False)
+        self.assertEqual(response.status_code, 401)
+
+    def test_update_atc_permission_denied(self):
+        response = self.put_json_data(reverse("osquery_api:atc", args=[1]), {})
+        self.assertEqual(response.status_code, 403)
+
+    def test_update_atc_already_exists(self):
+        configuration, atc = self.force_configuration(force_atc=True)
+        configuration2, atc2 = self.force_configuration(force_atc=True)
+        self.set_permissions("osquery.change_automatictableconstruction")
+        data = {
+            "name": atc.name,
+            "description": "yolo changed",
+            "path": "/home/yolo/new",
+            "query": "select new from yolo;",
+            "table_name": "yolo_new",
+            "columns": ["un", "deux", "trois"],
+            "platforms": ["darwin"]
+        }
+        response = self.put_json_data(reverse("osquery_api:atc", args=[atc2.id]), data)
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.json(), {
+            "name": ["automatic table construction with this name already exists."]
+        })
+
+    def test_update_atc(self):
+        atc = self.force_atc()
+        self.set_permissions("osquery.change_automatictableconstruction")
+        data = {
+            "name": "yolo",
+            "description": "yolo changed",
+            "path": "/home/yolo/new",
+            "query": "select new from yolo;",
+            "table_name": "yolo_new",
+            "columns": ["un", "deux", "trois"],
+            "platforms": ["darwin"]
+        }
+        response = self.put_json_data(reverse("osquery_api:atc", args=[atc.id]), data)
+        self.assertEqual(response.status_code, 200)
+        atc.refresh_from_db()
+        self.assertEqual(response.json(), {
+            "platforms": ["darwin"],
+            "updated_at": atc.updated_at.isoformat(),
+            "columns": ["un", "deux", "trois"],
+            "id": atc.id,
+            "created_at": atc.created_at.isoformat(),
+            "table_name": "yolo_new",
+            "query": "select new from yolo;",
+            "description": "yolo changed",
+            "path": "/home/yolo/new",
+            "name": atc.name
+        })
+        self.assertEqual(response.data, {
```

Comment:
testing response.data and response.json is a bit the same. The response and the object must be tested.